### PR TITLE
fluent bit: Add support for outbound http proxy

### PR DESCRIPTION
**Pull request by SanyaKochhar**
**Description**: Adds support for outbound proxy with Fluent Bit. Updates Fluent Bit to v1.6.4.

**Affected area**:

- New Functionality      [x]
- Documentation          [x]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
no, no

### CODE REVIEW

File: docs/patterns/observability/logs.md [hunk 1]
```diff
@@ -57,3 +57,24 @@ Once logs have been sent to Log Analytics, that can also be consumed by Applicat
 
 You can now interact with your logs in either of these instances.
 
+
+### Configuring Outbound Proxy Support for Fluent Bit
+If you require outbound proxy support:
```
Comment (michelleN) [inline docs/patterns/observability/logs.md:62]: Could you add a description around why someone would want or require outbound proxy support for fluent bit?

**Review by michelleN — CHANGES_REQUESTED** [review]

File: docs/patterns/observability/logs.md [hunk 2]
```diff
@@ -57,3 +57,24 @@ Once logs have been sent to Log Analytics, that can also be consumed by Applicat
...
+
+1. In `values.yaml`:
+    
+    a) Change `enableProxySupport` to `true`
+
+    b) Update the httpProxy and httpsProxy values to `"http://<host>:<port>"`
+
+2. In `/charts/osm/templates` create a new secret `proxy-config.yaml`:
```
Comment (ritazh) [inline docs/patterns/observability/logs.md:70]: This proxy-config.yaml can be created as part of the chart conditional on `enableProxySupport`

File: docs/patterns/observability/logs.md [hunk 3]
```diff
@@ -57,3 +57,24 @@ Once logs have been sent to Log Analytics, that can also be consumed by Applicat
...
 
+
+### Configuring Outbound Proxy Support for Fluent Bit
+If you require outbound proxy support:
+
+1. In `values.yaml`:
+    
+    a) Change `enableProxySupport` to `true`
+
+    b) Update the httpProxy and httpsProxy values to `"http://<host>:<port>"`
```
Comment (ritazh) [inline docs/patterns/observability/logs.md:68]: Might be good to also add a note here that fluent-bit `1.6.4` is required for this feature to work.

File: docs/patterns/observability/logs.md (hunk 1 above)
Comment (SanyaKochhar) [reply docs/patterns/observability/logs.md]: @ritazh I've added a line on this, could you take a look?

File: docs/patterns/observability/logs.md [hunk 4]
```diff
@@ -57,3 +57,12 @@ Once logs have been sent to Log Analytics, that can also be consumed by Applicat
 
 You can now interact with your logs in either of these instances.
 
+
+### Configuring Outbound Proxy Support for Fluent Bit
+You may require outbound proxy support if your traffic is configured to go through a proxy server. To enable this, make the following changes in `values.yaml`:
```
Comment (ritazh) [inline docs/patterns/observability/logs.md:62]: nit
```suggestion
You may require outbound proxy support if your egress traffic is configured to go through a proxy server. To enable this, make the following changes in `values.yaml`:
```

**Review by ritazh — DISMISSED** [review]
lgtm

File: docs/patterns/observability/logs.md [hunk 5]
```diff
@@ -57,3 +57,12 @@ Once logs have been sent to Log Analytics, that can also be consumed by Applicat
 
 You can now interact with your logs in either of these instances.
 
+
+### Configuring Outbound Proxy Support for Fluent Bit
+You may require outbound proxy support if your egress traffic is configured to go through a proxy server. To enable this, make the following changes in `values.yaml`:
```
Comment (michelleN) [inline docs/patterns/observability/logs.md:62]: Thanks!

**Review by michelleN — APPROVED** [review]
Thanks @SanyaKochhar ! Is there an e2e test scenario that validates the fluentbit integration somewhere?

**Comment by SanyaKochhar** [comment]
> Thanks @SanyaKochhar ! Is there an e2e test scenario that validates the fluentbit integration somewhere?

@michelleN There's this e2e for fluentbit integration https://github.com/openservicemesh/osm/blob/main/tests/e2e/e2e_fluentbit_test.go - but it doesn't test changes made in this PR

**Review by nshankar13 — APPROVED** [review]

**Comment by nshankar13** [comment]
lgtm

**Pull request merged**